handler-proxy/src/main/java/io/netty/handler/proxy/ProxyHandler.java

```diff
@@ -347,11 +349,12 @@ private void setConnectFailure(Throwable cause) {
                     exceptionMessage(cause.toString()), cause);
         }
 
-        if (connectPromise.tryFailure(cause)) {
+        if (!connectPromise.isDone()) {
             safeRemoveDecoder();
             safeRemoveEncoder();
 
             failPendingWrites(cause);
```

Comment:
consider extracting this into a private method and share code with `setConnectSuccess`'s `else` clause above.